ansible/library/bgp_route.py

```diff
@@ -0,0 +1,286 @@
+       Origin incomplete, localpref 100, valid, external, multipath
+       Last update: Fri Sep 22 06:33:37 2017
+
+    65200 62011 65501
+    10.0.0.9 from 10.0.0.9 (100.1.0.5)
+       Origin incomplete, localpref 100, valid, external, multipath, best
+       Last update: Fri Sep 22 06:33:37 2017
+
+
+
+ vtysh -c "show ip bgp neighbor 10.0.0.35 adv" 
+
+  BGP table version is 0, local router ID is 10.1.0.32
+  Status codes: s suppressed, d damped, h history, * valid, > best, = multipath,
+              i internal, r RIB-failure, S Stale, R Removed
+  Origin codes: i - IGP, e - EGP, ? - incomplete
+
+      Network          Next Hop            Metric LocPrf Weight Path
+   *> 10.1.0.32/32     10.0.0.34                 0         32768 i
+   *> 100.1.0.3/32     10.0.0.34                              0 65200 i
+   *> 100.1.0.5/32     10.0.0.34                              0 65200 i
+   *> 100.1.0.7/32     10.0.0.34                              0 65200 i
+   *> 100.1.0.9/32     10.0.0.34                              0 65200 i
+   *> 100.1.0.11/32    10.0.0.34                              0 65200 i
+   *> 100.1.0.13/32    10.0.0.34                              0 65200 i
+   *> 100.1.0.15/32    10.0.0.34                              0 65200 i
+   *> 100.1.0.17/32    10.0.0.34                              0 64001 i
+   *> 192.168.10.97/32 10.0.0.34                              0 65200 62011 65507 ?
+   *> 192.168.10.112/32
+                       10.0.0.34                              0 65200 62011 65508 ?
+   *> 192.168.10.113/32
+                       10.0.0.34                              0 65200 62011 65508 ?
+   *> 192.168.10.128/32
+                       10.0.0.34                              0 65200 62011 65509 ?
+   *> 192.168.10.129/32
+                       10.0.0.34                              0 65200 62011 65509 ?
+   *> 192.168.10.144/32
+                       10.0.0.34                              0 65200 62011 65510 ?
+
+'''
+
+
+### TODO: ipv6 route entries parsing is not fully working, need continue working on ipv6 ###
+
+
+class BgpRoutes(object):
+    '''
+        parsing bgp routing information
+    '''
+    def __init__(self):
+        self.facts = defaultdict(dict)
+        return
+
+    def get_facts(self):
+        return self.facts
+
+    def parse_bgp_rt_adv(self, cmd, cmd_result):
+        '''
+        parse BGP routing facts of neighbor advertised routes
+        '''
+        regex_neighbor = re.compile('nei.*\s([0-9a-fA-F.:]+)\s.*')
+        #regix_iprange = re.compile('[0-9a-fA-F.:]+\/\d+')
+        if regex_neighbor.match(cmd):
+            neighbor = regex_neighbor.match(cmd).group(1)
+        else:
+            raise Exception('cannot find neighbor in "show ip  bgp ' + cmd + '"')
+        self.facts['bgp_rt_neiadv']['neighbor'] = neighbor
+        ### so far parsing by fixed length table
+        ### will improve this if find better way to parse it
+        header_position = [0, 3, 20, 40, 47, 54, 61]
```

Comment:
why not use regx split('\s+') into several fields for a line.

Reply:
well, not every entry has all fields, so cannot split('\s+')
like following,  loopback route has a matric of 0 which rest of the example lines do not have that field:
   Network            Next Hop            Metric LocPrf Weight Path
*> 0.0.0.0              10.0.0.32                                             0 65200 ?
*> 10.1.0.32/32     10.0.0.32                     0                         32768 i
*> 100.1.0.1/32     10.0.0.32                                             0 65200 i